cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/external/ExternalDataset.java

```diff
@@ -45,21 +48,25 @@ public ExternalDataset(Map<String, String> runtimeArgs) {
       this.outputFormatClassName != null ? runtimeArgs : Collections.<String, String>emptyMap();
   }
 
+  @ReadOnly
```

Comment:
why are these methods (this and the following 3) not @NoAccess? They simply query configuration, but they don't perform actual data access.